extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -40,17 +40,38 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		return allFiles;
 	}
 
-	async openNotebook(resource: vscode.Uri): Promise<void> {
+	async openNotebook(resource: string): Promise<void> {
 		try {
 			let doc = await vscode.workspace.openTextDocument(resource);
 			vscode.window.showTextDocument(doc);
 		} catch (e) {
 			vscode.window.showErrorMessage(localize('openNotebookError', 'Open file {0} failed: {1}',
-				resource.fsPath,
+				resource,
 				e instanceof Error ? e.message : e));
 		}
 	}
 
+	async openMarkdown(resource: string): Promise<void> {
```

Comment:
Any specific reason this is async? You're not awaiting anything so this can just be a normal function

Reply:
Whoops sorry, I had copied from openNotebook